src/main/java/com/cazsius/solcarrot/SOLCarrot.java

```diff
@@ -2,47 +2,51 @@
 
 import com.cazsius.solcarrot.communication.FoodListMessage;
 import com.cazsius.solcarrot.item.SOLCarrotItems;
+import com.cazsius.solcarrot.tracking.FoodList;
 import net.minecraft.resources.ResourceLocation;
-import net.minecraftforge.eventbus.api.SubscribeEvent;
-import net.minecraftforge.fml.common.Mod;
-import net.minecraftforge.fml.event.lifecycle.FMLCommonSetupEvent;
-import net.minecraftforge.network.NetworkRegistry;
-import net.minecraftforge.network.simple.SimpleChannel;
+import net.neoforged.api.distmarker.Dist;
+import net.neoforged.bus.api.IEventBus;
+import net.neoforged.bus.api.SubscribeEvent;
+import net.neoforged.fml.ModContainer;
+import net.neoforged.fml.common.EventBusSubscriber;
+import net.neoforged.fml.common.Mod;
+import net.neoforged.neoforge.attachment.AttachmentType;
+import net.neoforged.neoforge.network.event.RegisterPayloadHandlersEvent;
+import net.neoforged.neoforge.network.registration.PayloadRegistrar;
+import net.neoforged.neoforge.registries.DeferredRegister;
+import net.neoforged.neoforge.registries.NeoForgeRegistries;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
-import static net.minecraftforge.fml.common.Mod.EventBusSubscriber.Bus.MOD;
+import java.util.function.Supplier;
+
+import static net.neoforged.fml.common.EventBusSubscriber.Bus.MOD;
 
 @Mod(SOLCarrot.MOD_ID)
-@Mod.EventBusSubscriber(modid = SOLCarrot.MOD_ID, bus = MOD)
+@EventBusSubscriber(modid = SOLCarrot.MOD_ID, bus = MOD)
 public final class SOLCarrot {
 	public static final String MOD_ID = "solcarrot";
 	
 	public static final Logger LOGGER = LogManager.getLogger(MOD_ID);
-	
-	private static final String PROTOCOL_VERSION = "1.0";
-	public static SimpleChannel channel = NetworkRegistry.ChannelBuilder
-		.named(resourceLocation("main"))
-		.clientAcceptedVersions(PROTOCOL_VERSION::equals)
-		.serverAcceptedVersions(PROTOCOL_VERSION::equals)
-		.networkProtocolVersion(() -> PROTOCOL_VERSION)
-		.simpleChannel();
-	
+
+	private static final DeferredRegister<AttachmentType<?>> ATTACHMENT_TYPES = DeferredRegister.create(NeoForgeRegistries.Keys.ATTACHMENT_TYPES, MOD_ID);
+	public static final Supplier<AttachmentType<FoodList>> FOOD_ATTACHMENT = ATTACHMENT_TYPES.register("food", () ->
+			AttachmentType.serializable(FoodList::new).build());
+
 	public static ResourceLocation resourceLocation(String path) {
-		return new ResourceLocation(MOD_ID, path);
+		return ResourceLocation.fromNamespaceAndPath(MOD_ID, path);
 	}
-	
```

Comment:
I thought I had some sort of editor config committed to the repo, but maybe it doesn't cover keeping indents on empty lines? I would prefer that, but it's not a big deal.

Reply:
My intelliJ keeps removing them upon trying to save :(